osu.Server.Spectator/Database/DatabaseAccess.cs

```diff
@@ -382,6 +402,78 @@ public async Task<IEnumerable<multiplayer_room>> GetActiveDailyChallengeRoomsAsy
                 + "AND `ends_at` > NOW()");
         }
 
+        public async Task<(long roomID, long playlistItemID)?> GetMultiplayerRoomIdForScoreAsync(long scoreId)
+        {
+            var connection = await getConnectionAsync();
+
+            return await connection.QuerySingleOrDefaultAsync<(long, long)?>(
+                "SELECT `multiplayer_playlist_items`.`room_id`, `multiplayer_playlist_items`.`id` "
+                + "FROM `multiplayer_score_links` "
+                + "JOIN `multiplayer_playlist_items` "
+                + "ON `multiplayer_score_links`.`playlist_item_id` = `multiplayer_playlist_items`.`id` "
+                + "WHERE `multiplayer_score_links`.`score_id` = @scoreId",
+                new { scoreId = scoreId });
+        }
+
+        public async Task<MultiplayerPlaylistItemStats[]> GetMultiplayerRoomStatsAsync(long roomId)
+        {
+            var connection = await getConnectionAsync();
+
+            long[] playlistItemIds = (await GetAllPlaylistItemsAsync(roomId)).Select(item => item.id).ToArray();
+            var result = new MultiplayerPlaylistItemStats[playlistItemIds.Length];
+
+            for (int i = 0; i < playlistItemIds.Length; ++i)
+            {
+                long[] totalScores = (await connection.QueryAsync<long>(
```

Comment:
I'm not sure how well this will scale, but I think we can go life with it and deal with that later.

Non-representative tests as we'll likely see a lot more usage than most existing playlists:

```sql
select playlist_item_id, count(score_id) from multiplayer_score_links group by playlist_item_id order by
                 -> count(score_id) desc limit 10;
+------------------+-----------------+
| playlist_item_id | count(score_id) |
+------------------+-----------------+
| 5415805          | 1090            |
| 7091206          | 353             |
| 5415807          | 321             |
| 5827890          | 291             |
| 4350065          | 280             |
| 4296355          | 279             |
| 5415798          | 253             |
| 5344648          | 250             |
| 7091692          | 249             |
| 4296362          | 242             |
+------------------+-----------------+

-- cold query (includes japan-us roundtrip)
select sum(total_score) from multiplayer_score_links join scores on (score_id = scores.id) where playlist_item_id = 5415805;
+------------------+
| sum(total_score) |
+------------------+
| 32986707         |
+------------------+

1 row in set
Time: 0.504s

-- warm query (includes japan-us roundtrip)
select sum(total_score) from multiplayer_score_links join scores on (score_id = scores.id) where playlist_item_id = 5415805;
+------------------+
| sum(total_score) |
+------------------+
| 32986707         |
+------------------+

1 row in set
Time: 0.130s
```

It shouldn't be a huge reach to cache these values if we need to, as well.

Reply:
If it's any solace the intention is that this will only be requested whenever someone enters the daily challenge screen.

Caching can 100% be done even now. It's non-critical data, nobody is going to care about perfect accuracy of this.

Reply:
Yep, aware of that. I'm also against caching until we need to.